### PR TITLE
Dev: Remove the `Honeypot` and `I18n` ignore errors in phpstan-baseline.php

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
See #7731 

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by datamweb — APPROVED** [review]

File: system/Honeypot/Exceptions/HoneypotException.php [hunk 1]
```diff
@@ -16,22 +16,22 @@
 
 class HoneypotException extends ConfigException implements ExceptionInterface
 {
-    public static function forNoTemplate()
+    public static function forNoTemplate(): self
```
Comment (kenjis) [inline system/Honeypot/Exceptions/HoneypotException.php:19]: This is a breaking change.
https://3v4l.org/GGS0f
Please use doc comment.

Comment (ping-yee) [reply system/Honeypot/Exceptions/HoneypotException.php]: Done.

**Review by kenjis — APPROVED** [review]
Thank you!

**Comment by kenjis** [comment]
Please rebase to resolve conflict.

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
@ping-yee Thank you!

**Pull request merged**